internal/receiver/awscontainerinsightreceiver/ecs/ecs_util.go

```diff
@@ -0,0 +1,127 @@
+package ecs
+
+import (
+	"fmt"
+	common "github.com/aws-observability/aws-otel-collector/internal/aws/containerinsightcommon"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+	"reflect"
+)
+
+func insertMetrics(key string, value interface{}, ilms pdata.InstrumentationLibraryMetricsSlice, metricType string, timestamp pdata.Timestamp, logger *zap.Logger) {
```

Comment:
Looks like there is a lot of duplicated code with `aws/containerinsightcommon/utils.go`. Can you create the ecs metrics not in OpenTelemetry format? You can just use map[string]interface{} to represent metrics values and then use map[string]string to represent the labels. In my opinion, I think it is easy to do metric decoration with those simple structures.